cni-plugin/deployment/scripts/install-cni.sh

```diff
@@ -55,94 +55,133 @@ CONTAINER_CNI_BIN_DIR=${CONTAINER_CNI_BIN_DIR:-/opt/cni/bin}
+    while read -r -d $'\0' file; do
+      echo "Removing linkerd-cni config from $file"
+      cni_data=$(jq 'del( .plugins[]? | select( .type == "linkerd-cni" ))' "$file")
+      # TODO (matei): we should write this out to a temp file and then do a `mv`
+      # to be atomic. 
+      echo "$cni_data" > "$file"
+    done
 
-    if [ "${CNI_CONF_PATH}" = "${CONTAINER_MOUNT_PREFIX}${DEST_CNI_NET_DIR}/01-linkerd-cni.conf" ]; then
-      rm -f "${CNI_CONF_PATH}"
-    fi
+  # Check whether configuration file has been created by our own cni plugin
+  # and if so, rm it.
+  if [ -e "${DEFAULT_CNI_CONF_PATH}" ]; then
+    echo "Cleaning up ${DEFAULT_CNI_CONF_PATH}"
+    rm -f "${DEFAULT_CNI_CONF_PATH}"
   fi
-  if [ -e "${CONTAINER_MOUNT_PREFIX}${DEST_CNI_NET_DIR}/${KUBECONFIG_FILE_NAME}" ]; then
-    echo "Removing linkerd-cni kubeconfig: ${CONTAINER_MOUNT_PREFIX}${DEST_CNI_NET_DIR}/${KUBECONFIG_FILE_NAME}"
-    rm -f "${CONTAINER_MOUNT_PREFIX}${DEST_CNI_NET_DIR}/${KUBECONFIG_FILE_NAME}"
+
+  # Remove binary and kubeconfig file
+  if [ -e "${HOST_CNI_NET}/${KUBECONFIG_FILE_NAME}" ]; then
+    echo "Removing linkerd-cni kubeconfig: ${HOST_CNI_NET}/${KUBECONFIG_FILE_NAME}"
+    rm -f "${HOST_CNI_NET}/${KUBECONFIG_FILE_NAME}"
   fi
   if [ -e "${CONTAINER_MOUNT_PREFIX}${DEST_CNI_BIN_DIR}"/linkerd-cni ]; then
     echo "Removing linkerd-cni binary: ${CONTAINER_MOUNT_PREFIX}${DEST_CNI_BIN_DIR}/linkerd-cni"
     rm -f "${CONTAINER_MOUNT_PREFIX}${DEST_CNI_BIN_DIR}/linkerd-cni"
   fi
+
   echo 'Exiting.'
+  exit 0
 }
 
 # Capture the usual signals and exit from the script
-trap cleanup EXIT
 trap 'echo "SIGINT received, simply exiting..."; cleanup' INT
 trap 'echo "SIGTERM received, simply exiting..."; cleanup' TERM
 trap 'echo "SIGHUP received, simply exiting..."; cleanup' HUP
 
-# Place the new binaries if the mounted directory is writeable.
-dir="${CONTAINER_MOUNT_PREFIX}${DEST_CNI_BIN_DIR}"
-if [ ! -w "${dir}" ]; then
-  exit_with_error "${dir} is non-writeable, failure"
-fi
-for path in "${CONTAINER_CNI_BIN_DIR}"/*; do
-  cp "${path}" "${dir}"/ || exit_with_error "Failed to copy ${path} to ${dir}."
-done
+# Install CNI bin will copy the linkerd-cni binary on the host's filesystem
+install_cni_bin() {
+    # Place the new binaries if the mounted directory is writeable.
+    dir="${CONTAINER_MOUNT_PREFIX}${DEST_CNI_BIN_DIR}"
+    if [ ! -w "${dir}" ]; then
+      exit_with_error "${dir} is non-writeable, failure"
+    fi
+    for path in "${CONTAINER_CNI_BIN_DIR}"/*; do
+      cp "${path}" "${dir}"/ || exit_with_error "Failed to copy ${path} to ${dir}."
+    done
+
+    echo "Wrote linkerd CNI binaries to ${dir}"
+}
+
+create_cni_conf() {
+  # Create temp configuration and kubeconfig files
+  #
+  TMP_CONF='/tmp/linkerd-cni.conf.default'
+  # If specified, overwrite the network configuration file.
+  # We are only interested in the side effect of parameter expansion here, hence ':'
```

Comment:
for my education, what's the side effect?

Reply:
So this is like a shorter version of this, as used for other vars at the top of this file? If so, it'd be nice to consolidate a single style :wink: 
```bash
CNI_NETWORK_CONFIG_FILE=${CNI_NETWORK_CONFIG_FILE:-}
```